### PR TITLE
34: call direct transform automatically for small N

### PR DESCRIPTION
**Pull request by michaelquellmalz**

### CODE REVIEW

File: kernel/nfft/nfft.c [hunk 1]
```diff
@@ -2478,6 +2478,12 @@ static void nfft_adjoint_1d_B(X(plan) *ths)
 
 void X(trafo_1d)(X(plan) *ths)
 {
+  if((ths->N[0] <= ths->m) || (ths->n[0] <= 2*ths->m+2))
```
Comment (jenskeiner) [inline kernel/nfft/nfft.c]: Ok, the first condition makes sense, the second condition is not too clear to me. Theoretically, only `n > N` is required, so `n = N + 1` should be a valid choice. In fact, I tested this by modifying the 2D case in `simple_test.c` where I set `m = 7`, `N[1] = 8`, and `n[1] = 9`. I didn't spot any issues when running the transformation. This doesn't mean that there isn't any place in our code (maybe dependent on window function or pre-computation method) where we assume `2 * m + 2 < n`, but it's not obvious to me.

File: kernel/nfft/nfft.c [hunk 2]
```diff
@@ -2478,6 +2478,12 @@ static void nfft_adjoint_1d_B(X(plan) *ths)
...
+  {
+    X(trafo_direct)(ths);
```
Comment (jenskeiner) [inline kernel/nfft/nfft.c]: Longer term, we should use a pointer to the correct trafo method in the plan. But that's a topic for NFFT 4.0. I suggest we create a branch where we can work on the change towards an opaque plan pointer. For some time, we will still need to support the 3.x.x line, but we should get started on the larger re-design.

File: kernel/nfft/nfft.c (hunk 1 above)
Comment (michaelquellmalz) [inline kernel/nfft/nfft.c]: Really? Because when I do the modifications you mentioned, `simple_test.exe` has segmentation fault.

Comment (tvolkmer) [inline kernel/nfft/nfft.c]: As far as I see from the `nfft.c` code, e.g., inside the function `static void nfft_trafo_1d_compute` there is a `for` loop:
```
  if (u < o)
  {
    for (l = 1, gj = g + u, (*fj) = (*psij++) * (*gj++); l <= 2*m+1; l++)
      (*fj) += (*psij++) * (*gj++);
  }
```
The arrays `g` and `g_hat` in the `nfft_plan` have size small `n` for the one-dimensional case. For `n < 2 * m + 1`, there might be a buffer overflow (invalid read for nfft_trafo, invalid write for nfft_adjoint).

Comment (jenskeiner) [inline kernel/nfft/nfft.c]: Ok, let's move forward on this one, but we should see if this limitations can be removed or not. At least the documentation would need to be updated if the code has to remain as it is. Maybe @skunis or @DanielPotts can help.

**Pull request merged**